### PR TITLE
[THORN-2126] upgrade to WildFly 14

### PR DESCRIPTION
**Pull request by Ladicek**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

### CODE REVIEW

**Comment by manovotn** [comment]
@Ladicek great job with WFLY-11373 workaround :)

**Comment by Ladicek** [comment]
Thanks @manovotn, though I kinda hoped I'd find a `Literal` for `@Resource` somewhere ;-)

**Comment by Ladicek** [comment]
Also could someone cleanup the disk on the Windows Jenkins agent? :-)

**Review by sberyozkin — COMMENTED** [review]
All of these changes look good to me. @kenfinnigan, please provide your review

File: docs/howto/pom.xml [hunk 1]
```diff
@@ -88,18 +91,6 @@
       </resource>
     </resources>
     <plugins>
-      <plugin>
```
Comment (kenfinnigan) [inline docs/howto/pom.xml:91]: Why was this removed?

File: fractions/javaee/jpa/pom.xml [hunk 2]
```diff
@@ -51,6 +51,11 @@
       <groupId>io.thorntail</groupId>
       <artifactId>ee</artifactId>
     </dependency>
+    <!-- WF14 -->
```
Comment (kenfinnigan) [inline fractions/javaee/jpa/pom.xml]: There's a hard dependency on Infinispan fraction for JPA? That's not good

File: build-parent/pom.xml [hunk 3]
```diff
@@ -248,6 +252,42 @@
         <version>${version.org.jboss.shrinkwrap.descriptors}</version>
       </dependency>
 
+      <dependency>
```
Comment (kenfinnigan) [inline build-parent/pom.xml:255]: Why are all these aether dependencies needed here? I didn't see them added anywhere

File: fractions/javaee/jpa/pom.xml (hunk 2 above)
Comment (sberyozkin) [reply fractions/javaee/jpa/pom.xml]: This is because of https://github.com/wildfly/wildfly/blob/14.0.1.Final/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/jpa/main/module.xml#L49 I believe

File: docs/howto/pom.xml (hunk 1 above)
Comment (Ladicek) [reply docs/howto/pom.xml:91]: Because each howto defines `thorntail-maven-plugin` on its own -- unfortunately, using a different `<id>` of the `<execution>`. So the Thorntail Maven plugin's `package` goal used to be invoked twice.

File: build-parent/pom.xml (hunk 3 above)
Comment (Ladicek) [reply build-parent/pom.xml:255]: Hmm, I don't recall adding this. I'll check.

File: fractions/javaee/jpa/pom.xml (hunk 2 above)
Comment (Ladicek) [reply fractions/javaee/jpa/pom.xml]: The Hibernate+Infinispan integration has moved from Hibernate to Infinispan, that might be the reason. But I'll look deeper.

Comment (Ladicek) [reply fractions/javaee/jpa/pom.xml]: Unfortunately it seems it really is required. Otherwise I'm getting

```
    "WFLYCTL0412: Required services that are not installed:" => ["org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity"],
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [
        "jboss.persistenceunit.\"CdiJpaJtaTest.jar#MyPU\".__FIRST_PHASE__ is missing [org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity]",
        "jboss.persistenceunit.\"CdiJpaJtaTest.jar#MyPU\" is missing [org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity]"
    ]
```

That is, the persistence unit service (from the JPA subsystem) depends on the Hibernate cache services (from the Infinispan subsystem).

Not sure how it worked before.

File: build-parent/pom.xml (hunk 3 above)
Comment (Ladicek) [reply build-parent/pom.xml:255]: @sberyozkin It seems you added this, do you recall why?

Comment (Ladicek) [reply build-parent/pom.xml:255]: Got it. Previously, Eclipse Aether was managed in `wildfly-core-parent` (which we import in `<dependencyManagement>`), but that's no longer the case. So we have to manage it ourselves.

**Comment by Ladicek** [comment]
Pushed a commit that addresses the remaining WF14 pieces. Unfortunately, it seems JPA having a hard dependency on Infinispan is now needed.

File: build-parent/pom.xml (hunk 3 above)
Comment (sberyozkin) [reply build-parent/pom.xml:255]: Yes, that is exactly why I added them, sorry, forgot I did it

Comment (kenfinnigan) [reply build-parent/pom.xml:255]: Ok, thanks for clarification

File: fractions/javaee/jpa/pom.xml (hunk 2 above)
Comment (kenfinnigan) [reply fractions/javaee/jpa/pom.xml]: Well that's going to blow up uber jar sizes.

Oh well

File: docs/howto/pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply docs/howto/pom.xml:91]: Makes sense, thanks

**Review by kenfinnigan — DISMISSED** [review]

File: fractions/javaee/jpa/pom.xml (hunk 2 above)
Comment (Ladicek) [reply fractions/javaee/jpa/pom.xml]: It does :-( The `jpa-jaxrs-cdi-jta` example has 110 MB with 2.2.1 and 126 MB now.

**Comment by Ladicek** [comment]
Linux build is green, yay! It would be great if we could get the disk cleaned up on the Windows agent, as getting a Windows build would be good for a change of this size.

**Comment by kenfinnigan** [comment]
retest this please

**Review by kenfinnigan — APPROVED** [review]
